Add contributing.md

gurukamath opened the pull request:
(closes #480 )

### What was wrong?
No documentation on the development tools is currently available

Related to Issue #480 

### How was it fixed?
Created `contributing.md`, moved relevant sections from `README.md` to it and added the missing developer tool documentation.

#### Cute Animal Picture

![Put a link to a cute animal picture inside the parenthesis-->](https://upload.wikimedia.org/wikipedia/commons/4/4f/Gálapagos_Penguins_Near_Isabela_Island.jpg)

SamWilsn commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+ 2. Any absolute package imports from other forks eg. in trie.py
+ 3. Package names under setup.cfg
+ 4. Add the new fork to the monkey_patch() function in src/ethereum_optimized/__init__.py
+ 5. Adjust the underline in fork/__init__.py
```
```suggestion
 5. Adjust the underline in `fork/__init__.py`
```

gurukamath replied:
Updated

SamWilsn reviewed (commented):
- Should mention that all code must be licensed under CC0.
 - Should we have a section on naming things? One suggestion:
   - Don't use EIP numbers in identifiers, and try to pick descriptive names instead.

gurukamath commented:
> * Should mention that all code must be licensed under CC0.
> * Should we have a section on naming things? One suggestion:
>   
>   * Don't use EIP numbers in identifiers, and try to pick descriptive names instead.

Added license to Readme. Added naming to the existing spelling section

gurukamath commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+In particular, we appreciate support in the following areas:
+
+- Reporting issues
+- Fixing and responding to [issues](https://github.com/ethereum/execution-specs/issues), especially those tagged as [good first issue](https://github.com/ethereum/execution-specs/labels/good%20first%20issue) which are meant as introductory issues for external contributors.
```
Updated

gurukamath commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+ * [PyPy 7.3.x](https://www.pypy.org/).
+ * `geth` installed and present in `$PATH`
+
+These version ranges are necessary because, at the time of writing, PyPy is only compatible with Python 3.7.
```
Updated. Please check.

gurukamath commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+
+Running the tests necessary to merge into the repository requires:
+
+ * Python 3.8.x (not 3.9 or later), and
```
Updated

gurukamath commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+# Contribution Guidelines
+
+Help is always welcome and there are plenty of options to contribute to EELS.
```
Updated

gurukamath commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+
+## CLI Utilities `ethereum_spec_tools`
+
+The execution specs repository has various CLI utilities that can help in the development process.
```
Updated

gurukamath commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,155 @@
+```
+
+The following will have to however, be updated manually
+ 1. The fork number and MAINNET_FORK_BLOCK in __init__.py
````
Updated

gurukamath commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+
+The following will have to however, be updated manually
+ 1. The fork number and MAINNET_FORK_BLOCK in __init__.py
+ 2. Any absolute package imports from other forks eg. in trie.py
```
Updated

gurukamath commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+The following will have to however, be updated manually
+ 1. The fork number and MAINNET_FORK_BLOCK in __init__.py
+ 2. Any absolute package imports from other forks eg. in trie.py
+ 3. Package names under setup.cfg
```
Updated

gurukamath commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+ 1. The fork number and MAINNET_FORK_BLOCK in __init__.py
+ 2. Any absolute package imports from other forks eg. in trie.py
+ 3. Package names under setup.cfg
+ 4. Add the new fork to the monkey_patch() function in src/ethereum_optimized/__init__.py
```
Updated

gurukamath commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,155 @@
+- Improving the documentation.
+
+
+For details about the Execution Specifications usage and building, please refer the [README](https://github.com/ethereum/execution-specs/blob/master/README.md#usage)
```
Updated
